ql/src/java/org/apache/hadoop/hive/ql/io/CombineHiveRecordReader.java

```diff
@@ -88,14 +90,26 @@ public CombineHiveRecordReader(InputSplit split, Configuration conf,
       }
     }
 
+    FileSplit inputSplit;
     // create a split for the given partition
-    FileSplit fsplit = new FileSplit(hsplit.getPaths()[partition], hsplit
-        .getStartOffsets()[partition], hsplit.getLengths()[partition], hsplit
-        .getLocations());
+    if (inputFormat instanceof CombineHiveInputFormat.MergeSplit) {
+      if (mrwork == null) {
+        mrwork = (MapWork) Utilities.getMergeWork(jobConf);
+        if (mrwork == null) {
```

Comment:
can it be null?

Reply:
It can be null, hence we do Utilities.getMapWork()